extra/src/main/java/aima/extra/probability/ProbabilityNumber.java

```diff
@@ -0,0 +1,40 @@
+package aima.extra.probability;
+
+import java.math.BigInteger;
+
+/**
+ * This interface is implemented by the various ProbabilityNumber
+ * implementations.
+ * 
+ * @author Nagaraj Poti
+ *
+ */
+public interface ProbabilityNumber {
```

Comment:
I have made some significant changes as per the updated requirements. I have added tests for all the four implementations as well.